build/litsearch/getInternalData.py

```diff
@@ -0,0 +1,233 @@
+            if len(dateParts) == 3:
+                return "-".join([str(dateParts[1]), str(dateParts[2]), str(dateParts[0])])
+            elif len(dateParts) == 2:
+                return "-".join([str(dateParts[1]), str(dateParts[0])])
+            elif len(dateParts) == 1:
+                return str(dateParts[0])
+            else:
+                return
+        except:
+            return
+
+    citationsDF.date = citationsDF.apply(dateStringFromDateParts, axis=1)
+
+    citationsDF.set_index("doi", inplace=True)
+    return citationsDF
+
+
+def mergePaperDataFrames(dataFramesList):
+    """ Combine a list of paper dataframes into one.
+
+    Each data frame should have the doi as index.
+    Can have title and date columns; the tile and date from the first dataframe in the list will be kept in case of conflict.
+    Any additional columns unique to dataset will be kept.
+    """
+    print(sum([len(df) for df in dataFramesList]), "total items to merge")
+
+    # Add "_#" to title and date columns of all but the first df
+    for i in range(len(dataFramesList)):
+        if i == 0: continue
+        dataFramesList[i] = dataFramesList[i].rename(columns={"title": ("title_" + str(i)), "date": ("date_" + str(i))})
+
+    # Seperate into items with and without valid DOIs
+    dataFramesList_DOI = []
+    dataFramesList_NoDOI = []
+    for df in dataFramesList:
+        validDOI = [str(index).startswith("10") for index in df.index]
+        invalidDOI = [False if val else True for val in validDOI]
+        dataFramesList_DOI.append(df[validDOI])
+        dataFramesList_NoDOI.append(df[invalidDOI])
+
+    # Merge on DOIs
+    mergedOnDOI = pd.concat(dataFramesList_DOI, axis=1, sort=False)
+    mergedOnDOI['doi'] = mergedOnDOI.index
+
+    # TODO: merge on titles
+
+    # Add in the items that didn't have a DOI
+    dfsToMerge = dataFramesList_NoDOI
+    dfsToMerge.append(mergedOnDOI)
+    merged = pd.concat(dfsToMerge, axis=0, ignore_index=True, sort=False)
+
+    # Combine the title and date info from duplicate columns
+    for i in range(len(dataFramesList)):
+        if i == 0: continue
+        for col in ['title', 'date']:
+            secondaryCol = col + "_" + str(i)
+            if secondaryCol in merged.columns:
+                merged[col] = merged[col].combine_first(df[secondaryCol])
+                merged.drop(secondaryCol, axis=1, inplace=True)
+
+    return merged
+
+
+# Main
+print("\n -- getting issues data --")
+issuesData = getIssuesData()
+print(len(issuesData), "'New Paper' issues")
+print("\n -- getting citations data --")
+citationsData = getCitationsData()
+print(len(citationsData), "citations in the covid19-review paper")
+# TODO: pullRequestsData = getPullRequestData()
+
+print("\n -- merging the data --")
+combinedData = mergePaperDataFrames([citationsData, issuesData])
+print(len(combinedData), "total items after merge")
+
+
+combinedDataFilePath = "./output/sources_cross_reference.tsv"
```

Comment:
Storing this in the `output` directory of the `output` branch makes sense to me.  Committing files back to the `master` branch gets messy.

If later we decide to include this as a new appendix, we could move it to the versioned website directories in the `gh-pages` branch.